### PR TITLE
Cleanup dev docs and split up redundant scripts

### PR DESCRIPTION
**Pull request by johscheuer**
This PR cleanups the dev docs to make them work with a more recent minikube version. I also split up the `start_{containerd,docker}.sh` script to only bootstrap the environment and added the `run_e2e_tests.sh` to run the e2e tests locally.

### CODE REVIEW

File: docs/developing.md [hunk 1]
```diff
@@ -86,12 +90,15 @@ In order to test newly build runner images we need to use the [Minikube Docker d
...
 ```
 
-And also deploy a local docker regisrty:
+And also deploy a local docker registry:
 
 ```bash
 kubectl apply -f local_dev/docker-registry.yml
 ```
 
+Now you need to add the minikube ip (you can get it with `minikube ip`) to the insecure regsitries of the [Docker client](https://docs.docker.com/registry/insecure/).
```
Comment (hacker-h) [inline docs/developing.md]: regsitries -> registries

File: local_dev/run_e2e_tests.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#use local docker daemon to build
+unset DOCKER_TLS_VERIFY
+unset DOCKER_HOST
+unset DOCKER_CERT_PATH
+unset DOCKER_API_VERSION
+
+ILLUMNATIO_IMAGE="$(minikube ip):5000/illuminatio-runner:dev"
+
+docker build -t "${ILLUMNATIO_IMAGE}" -f illuminatio-runner.dockerfile .
```
Comment (hacker-h) [inline local_dev/run_e2e_tests.sh]: ILLUMNATIO_IMAGE -> ILLUMINATIO_IMAGE

File: docs/developing.md (hunk 1 above)
Comment (johscheuer) [reply docs/developing.md]: Fixed, thanks!

File: .travis.yml [hunk 3]
```diff
@@ -154,13 +154,10 @@ jobs:
...
       # create local registry
       - kubectl apply -f local_dev/docker-registry.yml
-      # build and push image to local registry as soon as it is ready
-      - docker build -t "localhost:5000/illuminatio-runner:dev" -f illuminatio-runner.dockerfile .
-      - until docker push "localhost:5000/illuminatio-runner:dev"; do sleep 1; done
 
       script:
       # run illuminatio e2e tests
-      - python setup.py test --addopts="-m e2e"
+      - ./local_dev/run_e2e_tests.sh
```
Comment (hacker-h) [inline .travis.yml]: is this a workaround until https://github.com/inovex/illuminatio/issues/29 is done?

**Pull request merged**